tests/components/sungrow/test_init.py

```diff
@@ -0,0 +1,99 @@
+"""Test the Sungrow Solar Energy sensor."""
+from unittest.mock import patch
+
+from homeassistant.components.sungrow.config_flow import CannotConnect
+from homeassistant.components.sungrow.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from . import MockClient, create_entry, inverter_data
+
+
+async def test_setup(hass: HomeAssistant) -> None:
+    """Test unload."""
+    entry = create_entry(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert entry.state == ConfigEntryState.LOADED
+
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entry.state is ConfigEntryState.NOT_LOADED
+    assert not hass.data.get(DOMAIN)
+
+
+async def test_device_info(hass: HomeAssistant) -> None:
+    """Test device info."""
+    entry = create_entry(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    device_registry = dr.async_get(hass)
+    await hass.async_block_till_done()
+    device = device_registry.async_get_device({(DOMAIN, entry.entry_id)})
+    await hass.async_block_till_done()
+
+    assert device is not None
+    assert device.configuration_url == "http://1.1.1.1"
+    assert device.entry_type is None
+    assert device.identifiers == {(DOMAIN, entry.entry_id)}
+    assert device.manufacturer == "Sungrow"
+    assert device.name == "Mock Title"
+
+
+async def test_device_data(hass: HomeAssistant) -> None:
+    """Test device data."""
+    with patch(
+        "homeassistant.components.sungrow.SungrowData.update",
+        return_value=inverter_data,
+    ) as mock_client:
+        entry = create_entry(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+        await hass.data[DOMAIN][entry.entry_id].async_refresh()
+
+        assert (
+            hass.data[DOMAIN][entry.entry_id].data["4990 ~ 4999 - Serial number"]
```

Comment:
Consider defining these special strings as constants.

Reply:
I did that